app/assets/javascripts/discourse/app/services/composer.js

```diff
@@ -342,16 +334,24 @@ export default class ComposerService extends Service {
     return whisperer && modelAction === Composer.REPLY;
   }
 
-  _setupPopupMenuOption(callback) {
-    let option = callback(this);
+  _setupPopupMenuOption(option) {
+    // Backwards compatibility support for when we used to accept a function
+    if (typeof option === "function") {
```

Comment:
Do we want to deprecate this old pattern?

Reply:
I added a note to remove this code when the `addToolbarPopupMenuOptionsCallback` plugin api is removed. The new API doesn't specify a function in the docs so it is unlikely devs will pass a function to it. Even if they do, we can consider it not supported.